src/Symfony/Component/Console/Parallelization/ProcessLauncher.php

```diff
@@ -0,0 +1,197 @@
+     *                                              each process receives
+     * @param LoggerInterface $logger               A logger for debug output
+     * @param callable        $callback             The callback that receives
+     *                                              the output of the child
+     *                                              processes
+     */
+    public function __construct(
+        string $command,
+        string $workingDirectory,
+        array $environmentVariables,
+        int $processLimit,
+        int $segmentSize,
+        LoggerInterface $logger,
+        callable $callback
+    ) {
+        $this->command = $command;
+        $this->workingDirectory = $workingDirectory;
+        $this->environmentVariables = $environmentVariables;
+        $this->processLimit = $processLimit;
+        $this->segmentSize = $segmentSize;
+        $this->logger = $logger;
+        $this->callback = $callback;
+    }
+
+    /**
+     * Runs child processes to process the given elements.
+     *
+     * @param string[] $elements The elements to process. None of the elements
+     *                           must contain newlines
+     */
+    public function run(iterable $elements): void
+    {
+        $currentInputStream = null;
+        $numberOfStreamedElements = 0;
+
+        foreach ($elements as $element) {
+            // Close the input stream if the segment is full
+            if (null !== $currentInputStream && $numberOfStreamedElements >= $this->segmentSize) {
+                $currentInputStream->close();
+
+                $currentInputStream = null;
+                $numberOfStreamedElements = 0;
+            }
+
+            // Wait until we can launch a new process
+            while (null === $currentInputStream) {
+                $this->freeTerminatedProcesses();
+
+                if (count($this->runningProcesses) < $this->processLimit) {
+                    // Start a new process
+                    $currentInputStream = new InputStream();
+                    $numberOfStreamedElements = 0;
+
+                    $this->startProcess($currentInputStream);
+
+                    break;
+                }
+
+                // 100ms
+                usleep(100000);
+            }
+
+            // Stream the data segment to the process' input stream
+            $currentInputStream->write($element."\n");
+
+            ++$numberOfStreamedElements;
+        }
+
+        if (null !== $currentInputStream) {
+            $currentInputStream->close();
+        }
+
+        while (count($this->runningProcesses) > 0) {
+            $this->freeTerminatedProcesses();
+
+            // 100ms
+            usleep(100000);
+        }
+    }
+
+    /**
+     * Starts a single process reading from the given input stream.
+     *
+     * @param InputStream $inputStream The input stream
+     */
+    private function startProcess(InputStream $inputStream): void
+    {
+        $process = new Process(
```

Comment:
That means each element to process will be executed in separated process, each of them will need to bootstrap application again and again. 
Can we use threads instead?